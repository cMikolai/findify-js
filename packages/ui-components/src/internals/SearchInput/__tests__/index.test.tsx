import * as React from 'react';
import { SearchInput, Props } from '..';

describe('<SearchInput />', () => {
  it('correctly renders itself', () => {
    const component = render(<SearchInput value="foo" placeholder="bar" />);
    expect(component).toMatchSnapshot();
  });

  it('passes current value to the onChange handler', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <SearchInput value="foo" placeholder="bar" onChange={onChange} />
    );
    const value = 'changed';
    wrapper.find('input').simulate('change', { target: { value } });
    expect(onChange).toBeCalledWith(value);
  });
});
