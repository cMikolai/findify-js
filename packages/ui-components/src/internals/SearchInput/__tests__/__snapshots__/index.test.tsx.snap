// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> correctly renders itself 1`] = `
<div
  class="wrap"
>
  <default
    class="icon"
    height="16"
    width="16"
  />
  <input
    class="input"
    placeholder="bar"
    type="search"
    value="foo"
  />
</div>
`;
